pkg/scheduler/scheduler.go

```diff
@@ -554,7 +554,19 @@ func (sched *Scheduler) Run(ctx context.Context) {
 // in-place podInformer.
 func NewInformerFactory(cs clientset.Interface, resyncPeriod time.Duration) informers.SharedInformerFactory {
 	informerFactory := informers.NewSharedInformerFactory(cs, resyncPeriod)
-	informerFactory.InformerFor(&v1.Pod{}, newPodInformer)
+	informer := informerFactory.InformerFor(&v1.Pod{}, newPodInformer)
+
+	// Dropping `.metadata.managedFields` to improve memory usage.
+	// The Extract workflow (i.e. `ExtractPod`) should be unused.
+	trim := func(obj interface{}) (interface{}, error) {
+		if accessor, err := meta.Accessor(obj); err == nil {
+			if accessor.GetManagedFields() != nil {
+				accessor.SetManagedFields(nil)
+			}
+		}
+		return obj, nil
+	}
+	informer.SetTransform(trim)
```

Comment:
Maybe just inject transformer for informerFactory?
`informerFactory := informers.NewSharedInformerFactoryWithOptions(cs, resyncPeriod, informers.WithTransform(trim))`